app/controllers/campaign_notes_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CampaignNotesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_campaign
+
+  def index
+    @notes = @campaign.comment_threads
+  end
+
+  def create
+    comment = Comment.build_from(@campaign, current_user.id, params[:note])
+    comment.save
+    redirect_to campaign_campaign_notes_path(@campaign)
+  end
+
+  private
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_campaign
+    @campaign = Campaign.find(params[:campaign_id])
+  end
+end
```

Comment:
I think we could likely have a polymorphic `comments` controller and a single set of views that work for any model we give `acts_as_commentable_with_threading` behavior to.

```ruby
# config/routes.rb
resources :campaigns do
  resources :comments
end
```